### PR TITLE
ISPN-8736 Rest authorization

### PR DESCRIPTION
**Pull request by tristantarrant**
- Replaces use of SecurityDomain with ServerAuthenticationProvider
- Use SecurityActions where appropriate
- Return 403 on unauthorized access

https://issues.jboss.org/browse/ISPN-8736

### CODE REVIEW

**Comment by gustavonalle** [comment]
There a failure on ```org.infinispan.server.test.security.rest.RESTBasicSecurityIT.testSecuredReadWriteOperations```

**Comment by tristantarrant** [comment]
run performance tests please
cs

**Comment by rvansa** [comment]
@tristantarrant Fyi, you can type just `cs-rest` instead of `cs` - this way you're running memcached and Hot Rod tests as well.

**Comment by tristantarrant** [comment]
@rvansa I made changes to how SecurityActions check whether to escalate privileges which affect all remote endpoints, so I want them all.

File: server/rest/src/test/java/org/infinispan/rest/security/AuthorizationTest.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            .send();
+
+      //then
+      ResponseAssertion.assertThat(put).isOk();
+
+      // when writer gets
+      ContentResponse get = request("default", "key", "writer", "password", HttpMethod.GET).send();
+
+      //then
+      ResponseAssertion.assertThat(get).isForbidden();
```
Comment (slaskawi) [inline server/rest/src/test/java/org/infinispan/rest/security/AuthorizationTest.java]: Could you please move those test cases into separate methods?

**Comment by tristantarrant** [comment]
This has a ~4% impact on non-secure REST, so I need to fix it further

**Comment by tristantarrant** [comment]
run performance tests please
cs-rest

File: server/rest/src/test/java/org/infinispan/rest/security/AuthorizationTest.java (hunk 1 above)
Comment (tristantarrant) [reply server/rest/src/test/java/org/infinispan/rest/security/AuthorizationTest.java]: Done

**Comment by tristantarrant** [comment]
run performance tests please
cs-rest

**Comment by tristantarrant** [comment]
Great, now we have authz checks in and a +3% speedup

**Comment by ryanemerson** [comment]
@tristantarrant the two test failures are related:

```
org.infinispan.server.test.security.rest.RESTBasicSecurityIT.testSecuredReadWriteOperations
org.infinispan.rest.logging.RestAccessLoggingTest.testRestAccessLog
```

**Comment by tristantarrant** [comment]
Closing until we move to Elytron

**Pull request closed without merging**